### PR TITLE
Fixes flashes in cyborg heads, buy random item and buy surplus crate

### PR DESCRIPTION
**Pull request by tortellinitony**

### CODE REVIEW

File: code/game/objects/items/robot/robot_parts.dm [hunk 1]
```diff
@@ -357,8 +357,8 @@
 		return ..()
 
 /obj/item/robot_parts/head/attackby(obj/item/W, mob/user, params)
-	if(istype(W, /obj/item/device/assembly/flash/handheld))
-		var/obj/item/device/assembly/flash/handheld/F = W
```
Comment (GuyonBroadway) [inline code/game/objects/items/robot/robot_parts.dm]: I FUCKING KNEW IT WAS SOME SNOWFLAKEY BULSHIT TEE GEE-FUCKING-WIZ DID!

**Comment by GuyonBroadway** [comment]
tl;dr THANKS TEE GEE!!

File: code/datums/uplink_item.dm [hunk 2]
```diff
@@ -1,1292 +0,0 @@
-var/list/uplink_items = list()
```
Comment (ZachDeava) [inline code/datums/uplink_item.dm]: Uh....what? Your fix is just to delete the entire list?

File: code/modules/uplink/uplink_item.dm [hunk 3]
```diff
@@ -1228,7 +1228,7 @@ var/list/uplink_items = list() // Global list so we only initialize this once.
 	cost = 0
 
 /datum/uplink_item/badass/random/spawn_item(turf/loc, obj/item/device/uplink/U)
-	var/list/uplink_items = get_uplink_items()
```
Comment (ZachDeava) [inline code/modules/uplink/uplink_item.dm]: Removing the call to set the list with no replacement?

File: code/datums/uplink_item.dm (hunk 2 above)
Comment (ZachDeava) [inline code/datums/uplink_item.dm]: Actually, I see this is made redundant by an exact duplicate in modules. This is fine.

**Pull request merged**